plugins/azure/frontdoor/frontDoorSecureOrigins.js

```diff
@@ -0,0 +1,89 @@
+    recommended_action: 'Ensure that Azure Front Door Standard and Premium profile origins are using private link.',
+    link: 'https://learn.microsoft.com/en-us/azure/frontdoor/origin-security?pivots=front-door-standard-premium&tabs=app-service-functions',
+    apis: ['profiles:list', 'afdOriginGroups:listByProfile', 'afdOrigin:listByOriginGroups'],
+
+    run: function(cache, settings, callback) {
+        const results = [];
+        const source = {};
+        const locations = helpers.locations(settings.govcloud);
+        async.each(locations.profiles, (location, rcb) => {
+            const profiles = helpers.addSource(cache, source,
+                ['profiles', 'list', location]);
+
+            if (!profiles) return rcb();
+
+            if (profiles.err || !profiles.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query Azure Front Door profiles: ' + helpers.addError(profiles), location);
+                return rcb();
+            }
+
+            if (!profiles.data.length) {
+                helpers.addResult(results, 0, 'No existing Azure Front Door profiles found', location);
+                return rcb();
+            }
+
+            var frontDoorProfile = false;
+            profiles.data.forEach(function(profile) {
+                if (!profile.id || profile.kind != 'frontdoor') return;
+
+                frontDoorProfile = true;
+                var insecureOriginGroupNames = {};
+                var originFound = false;
+
+                const afdOriginGroups = helpers.addSource(cache, source,
+                    ['afdOriginGroups', 'listByProfile', location, profile.id]);
+
+                if (!afdOriginGroups || afdOriginGroups.err || !afdOriginGroups.data) {
+                    helpers.addResult(results, 3,
+                        'Unable to query Azure Front Door origin groups: ' + helpers.addError(afdOriginGroups), location, profile.id);
+                } else if (!afdOriginGroups.data.length) {
+                    helpers.addResult(results, 0, 'Front Door profile does not have any origin groups', location, profile.id);
+                } else {
+                    afdOriginGroups.data.forEach(function(originGroup) {
+                        const afdOrigin = helpers.addSource(cache, source,
+                            ['afdOrigin', 'listByOriginGroups', location, originGroup.id]);
+
+                        if (!afdOrigin || afdOrigin.err || !afdOrigin.data) {
+                            helpers.addResult(results, 3,
+                                'Unable to query Azure Front Door origins: ' + helpers.addError(afdOrigin), location, profile.id);
+                        } else {
+                            originFound = true;
```

Comment:
Why we need this? We can use insecureOriginGroupNames

Reply:
if for some reason origins are not being fetched i.e. permission issue, it will go straight for insecureOriginGroupNames which will be empty in that case and it will generate false positive results